### PR TITLE
Migrate docker installation method

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Migrate docker installation method from package to repository

### CODE REVIEW

File: src/upgrades/upgrade_013.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  log "Failed to install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin."
+  exit 1
+fi
+# 3. Verify that the Docker Engine installation is successful by running the hello-world image.
+#docker run --rm hello-world && docker rmi hello-world
+
+# Add docker-compose alias of docker compose to the dappnode profile if is not already there and if docker compose is installed
+if [ -x "$(command -v docker-compose)" ]; then
+  if ! grep -q "alias docker-compose='docker compose'" /usr/src/dappnode/DNCORE/.dappnode_profile; then
+    log "Adding docker-compose alias to the dappnode profile"
```
Comment (pablomendezroyo) [inline src/upgrades/upgrade_013.sh:138]: Check if `docker compose` actually exists

**Comment by pablomendezroyo** [comment]
@3alpha @Marketen @dappnodedev please add your review, this is a critical implementation 🙏

File: src/scripts/docker_migration_apt.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  log "WARNING: Unattended upgrades is not enabled, skipping upgrade"
+  exit 0
+fi
+
+# Check if docker is installed via apt
+# The docker.list file is created by the docker installation script
+if [ -f /etc/apt/sources.list.d/docker.list ]; then
+  log "Docker is already installed via apt, skipping upgrade"
+  # TODO: check if docker is in unattended upgrades
+  exit 0
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:27]: Be careful here, with docker installed via apt you will not add it to the unattended upgrades if you exit here

File: src/scripts/docker_migration_apt.sh [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  OS=$DISTRIB_ID
+  VERSION=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+  OS=Debian
+  VERSION=$(cat /etc/debian_version)
+else
+  OS=$(uname -s)
+  VERSION=$(uname -r)
+fi
+
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:52]: Maybe you should stop installation if none of the above branches is `true`?

File: src/scripts/docker_migration_apt.sh [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  exit 0
+fi
+
+# Print OS and version for the migration
+log "OS: $OS ; Version: $VERSION ; Docker version: $DOCKER_VERSION"
+
+# Remove legacy docker packages
+for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do
+  log "Removing $pkg"
+  apt-get remove $pkg 2>&1 | tee -a ${LOG_FILE}
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:73]: Maybe we could also `prune`, not only `remove`

File: src/scripts/docker_migration_apt.sh [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+fi
+log "Install ca-certificates curl and gnupg"
+apt-get install -y ca-certificates curl gnupg 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to install ca-certofocates curl and gnupg."
+  exit 1
+fi
+# 2. Add Docker's official GPG key
+log "Add Docker's official GPG key"
+install -m 0755 -d /etc/apt/keyrings 2>&1 | tee -a ${LOG_FILE}
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:93]: What is being done here? Where is the key obtained from? I do not fully understand this

File: src/scripts/docker_migration_apt.sh [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+fi
+
+# 1. Update the apt package index:
+log "Update the apt packages again"
+apt-get update 2>&1 | tee -a ${LOG_FILE}
+if [ $? -ne 0 ]; then
+  log "Failed to update"
+  exit 1
+fi
+# IMPORTANT: This step MUST be skipped so unattended-upgrades will upgrade docker later on
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:129]: We must be sure that the unattended upgrades will install it. Maybe if it is not installed the upgrade is skipped

File: src/scripts/docker_migration_apt.sh [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+if [ -x "$(command -v docker-compose)" ]; then
+  if ! grep -q "alias docker-compose='docker compose'" /usr/src/dappnode/DNCORE/.dappnode_profile; then
+    log "Adding docker-compose alias to the dappnode profile"
+    echo "alias docker-compose='docker compose'" >>/usr/src/dappnode/DNCORE/.dappnode_profile
+    source /usr/src/dappnode/DNCORE/.dappnode_profile | tee -a ${LOG_FILE}
+    if [ $? -ne 0 ]; then
+      log "Failed to source dappnode profile."
+      exit 1
+    fi
+    # TODO: consider removing old docker-compose binary
```
Comment (dsimog01) [inline src/scripts/docker_migration_apt.sh:158]: You are right here. Do we know which of them will be used if both exist ( `docker-compose` alias and `docker-compose` old binary)?

**Review by dsimog01 — CHANGES_REQUESTED** [review]

File: src/scripts/docker_migration_apt.sh (hunk 5 above)
Comment (pablomendezroyo) [reply src/scripts/docker_migration_apt.sh:93]: see either docker install debian/ubuntu/raspbian repository instructions for further details 

https://docs.docker.com/engine/install/debian/

File: src/scripts/docker_migration_apt.sh (hunk 6 above)
Comment (pablomendezroyo) [reply src/scripts/docker_migration_apt.sh:129]: you can verify it with `unattended-upgrade -d`

**Comment by pablomendezroyo** [comment]
Depends on https://github.com/dappnode/DNP_DAPPMANAGER/issues/1592

File: src/scripts/docker_migration_apt.sh (hunk 3 above)
Comment (pablomendezroyo) [reply src/scripts/docker_migration_apt.sh:52]: already exiting in next step

File: src/scripts/docker_migration_apt.sh (hunk 4 above)
Comment (pablomendezroyo) [reply src/scripts/docker_migration_apt.sh:73]: following official docker instructs

**Pull request merged**